shopinvader_delivery_carrier_category_keep_carrier/services/cart.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import logging
+
+from odoo.addons.component.core import Component
+
+_logger = logging.getLogger(__name__)
+
+
+class CartService(Component):
+
+    _inherit = "shopinvader.cart.service"
+
+    def _check_carrier_unset(self, cart, params):
```

Comment:
@rousseldenis The method name is a little bit confusing or at least lack of meaning. (The same applies to the docstring) The purpose of this method is to know if the carrier should be reset on the cat when the shipping address change on the SO... `_reset_carrier_on_shipping_address_update`

Reply:
> @rousseldenis The method name is a little bit confusing or at least lack of meaning. (The same applies to the docstring) The purpose of this method is to know if the carrier should be reset on the cat when the shipping address change on the SO... `_reset_carrier_on_shipping_address_update`

It's a behaviour typically done in shopinvader when changing the shipping adress. So, a module outside shopinvader would have no utility. This method is a simple hook I've added but could have been done in _unset_carrier()